### DISCUSSION TITLE
Switch from October to Winter - Composer update : InteractsWithAuthentication

### DISCUSSION THREAD

**Discussion by m-paniez9292** [wintercms/winter#412] [Q&A] [answered]
### Winter CMS Build

Other (please specify below)

### PHP Version

8.0

### Database engine

MySQL/MariaDB

### Issue description

Hi all,
We have a little website on October and would use it to test the switch method to Winter.
The October version is 473 and we put composer.json to use the "Wizard Method Installation" switch method describe on https://wintercms.com/blog/post/how-start-using-winter-cms .
Composer version is 2
When we put "composer update", we have :
```
[RuntimeException]                                                                                                                  
  Could not scan for classes inside "tests/concerns/InteractsWithAuthentication.php" which does not appear to be a file nor a folder
```
This problem is appears just after modules install (previous lines) :
```
  - Installing winter/wn-system-module (dev-develop 3fab4e6): Extracting archive
76 package suggestions were added by new dependencies, use `composer suggest` to see details.
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Generating autoload files
```

On frontend, we have this message :
`Missing vendor files, try running "composer install" or use the Wizard installer. `
We do a new `composer update` but nothing change (frontend message and console error).

EDIT :
We follow https://github.com/wintercms/winter/discussions/393 and remove autoload-dev section, but we have another error :
```
> php artisan winter:version
Missing vendor files, try running "composer install" or use the Wizard installer.
Script php artisan winter:version handling the post-update-cmd event returned with error code 1
```
Previous lines are :
```
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Generating autoload files
75 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```
END EDIT

Thanks

### Steps to replicate

This website is recent (3 month ago) and it has only 1 page.
We just copy-paste your composer.json and run `composer update`

**Comment by bennothommo** [comment, marked as answer]
@m-paniez9292 did you run `composer update` after removing the `autoload-dev` section from `composer.json`?

**Reply by m-paniez9292** [reply]
Hi,

Sure, it's our edit reply

**Reply by bennothommo** [reply]
Sorry, it just said that you had removed the section, it didn't say that you re-ran the update.

Try deleting `composer.lock` and the `vendor` directly entirely, then re-run `composer update`.

**Reply by m-paniez9292** [reply]
We do your process and we have the same issue.
Not exactly the same : composer has installed 2 new packages (mustangostang/spyc and matomo/device-detector). After these 2 packages installation, we have the same error :
```
> php artisan winter:version
Missing vendor files, try running "composer install" or use the Wizard installer.
Script php artisan winter:version handling the post-update-cmd event returned with error code 1
```

**Reply by LukeTowers** [reply]
You need to update your bootstrap autoload.php file, it's in the migration instructions.

**Reply by m-paniez9292** [reply]
Re,
You are right.
I have another issue : I continue here or I post a new issue ?
Thanks

**Reply by LukeTowers** [reply]
Might as well continue here. You can also use our discord to get help.

**Reply by m-paniez9292** [reply]
Re,
composer update seems OK (no error).
When I'm going to our website, I have the message :
`Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.0.2".`
But our PHP is :
```
php -v
PHP 8.0.14 (cli) (built: Dec 20 2021 21:32:16) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.14, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.14, Copyright (c), by Zend Technologies
```

Thanks

**Reply by LukeTowers** [reply]
Check what version of php your webserver is using, it can be configured differently than your CLI

**Reply by m-paniez9292** [reply]
Re,
Ok, php upgraded to 8.1.1
But now, we have :
`Script php artisan winter:version handling the post-update-cmd event returned with error code 255`
and stack trace :
```
  During inheritance of ArrayAccess: Uncaught ErrorException: Return type of Illuminate\Http\Request::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool,  
   or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /home/www/vendor/laravel/framework/src/Illuminate/Http/Request.php:641                        
  Stack trace:                                                                                                                                                                                                
  #0 /home/www/vendor/laravel/framework/src/Illuminate/Http/Request.php(20): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()                                                    
  #1 /home/www/vendor/composer/ClassLoader.php(571): include('...')                                                                                                                             
  #2 /home/www/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile()                                                                                                            
  #3 /home/www/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/SetRequestForConsole.php(31): Composer\Autoload\ClassLoader->loadClass()                                            
  #4 /home/www/vendor/winter/storm/src/Foundation/Application.php(127): Illuminate\Foundation\Bootstrap\SetRequestForConsole->bootstrap()                                                       
  #5 /home/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(320): Winter\Storm\Foundation\Application->bootstrapWith()                                                 
  #6 /home/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\Foundation\Console\Kernel->bootstrap()                                                    
  #7 /home/www/artisan(35): Illuminate\Foundation\Console\Kernel->handle()                                                                                                                      
  #8 {main}
```
Sorry to disturb you with that.

Thanks

**Reply by LukeTowers** [reply]
Winter v1.0 and 1.1 don't support PHP 8.1 because Laravel v6 LTS doesn't support it. Use 8.0 for now and in a couple weeks Winter v1.2 should be available to use with support for Laravel v9 LTS and PHP 8.1

**Reply by m-paniez9292** [reply]
Re,
So, we downgrade PHP to 8.0.14.
`composer update` works well.
If we go to the website, we have this message : `make('Illuminate\Contracts\Http\Kernel'); $response = $kernel->handle( $request = Illuminate\Http\Request::capture() ); $response->send(); $kernel->terminate($request, $response); `

Thanks

**Reply by LukeTowers** [reply]
check what version of PHP your webserver is running by putting 
```php
phpinfo();
exit;
```
at the top of your index.php file.

**Reply by m-paniez9292** [reply]
Re,

All is ok.
Thanks for your time.
